ros_buildfarm/templates/doc/doc_independent_docker_job.xml.em

```diff
@@ -81,14 +87,18 @@
 @{
 import os
 doc_repository_name = os.path.splitext(os.path.basename(doc_repository_url))[0]
+
+if doc_repository_branch is None:
+  doc_repository_branch = ''
+else:
+  doc_repository_branch = '--no-single-branch -b ' + doc_repository_branch
 }@
 @(SNIPPET(
     'builder_shell',
     script='\n'.join([
         'echo "# BEGIN SECTION: Clone %s"' % doc_repository_name,
-        'python3 -u $WORKSPACE/ros_buildfarm/scripts/wrapper/git.py clone --depth 1 %s $WORKSPACE/repositories/%s' % (doc_repository_url, doc_repository_name),
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/wrapper/git.py clone --depth 1 %s %s $WORKSPACE/repositories/%s' % (doc_repository_url, doc_repository_branch, doc_repository_name),
```

Comment:
Don't you need a `-b` flag before the branch name?

Reply:
That gets added in the code above.  That is, if `doc_repository_branch` is *not* specified, then the `doc_repository_branch` string is empty and we get the default branch.  If `doc_repository_branch` *is* specified, then the `doc_repository_branch` string is `--no-single-branch -b <branch>`, and we get the requested branch.

Reply:
I'm somewhat leery of overwriting the `doc_repository_branch` as it means that once we've modified it to create this argument we no longer have access to the "raw" branch name should we need it. But consider this a nitpick.

Reply:
Yeah, that's fine.  I can rename the variable to something else.

Reply:
Fixed in 1b81fcd